src/eliminate_common_subexpression.cpp

```diff
@@ -30,9 +30,15 @@ void cse_range(module& p, Range&& r)
                 continue;
             if(*eq != *ins)
                 continue;
+            auto outputs = eq->outputs();
+            for(auto output : outputs)
+            {
+                if(not p.has_instruction(output))
+                    continue;
```

Comment:
This loop looks like it doesn't do anything.

Reply:
This skips over the replacement of instructions within submodules since it checks if the current module contains the output instruction or not.

Reply:
> This skips over the replacement of instructions

How? After the loop finishes it will call `replace_instruction` regardless of what `p.has_instruction(output)` returns.

Reply:
I see what you mean, I originally intended to skip the outer for loop, but this is not happening in the above change. For some reason the reordering of getting the outputs before calling replace is fixing the issue.